### PR TITLE
pass authprovider to buildkit session

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
Aligned compose build code with https://github.com/docker/buildx/blob/master/commands/build.go

**Related issue**
https://github.com/docker/compose-cli/issues/1545

### CODE REVIEW

**Comment by ndeloof** [comment]
@tonistiigi could you please confirm this is the correct way?
i.e. `GetDriver` already has an `Auth` parameter we pass docker config file, why do we also have to set this as `Session` option?

**Review by gtardif — APPROVED** [review]
LGTM, Just pending @tonistiigi confirmation

**Comment by ndeloof** [comment]
go.mod is killing me ...

**Comment by tonistiigi** [comment]
> i.e. GetDriver already has an Auth parameter we pass docker config file, why do we also have to set this as Session option?

Yes, both are needed. `GetDriver` is only used when pulling buildkit image itself for creating the builder container one time. That image is public but we pass auth to avoid registry side rate-limiting.

**Comment by ndeloof** [comment]
dependency issue:
buildx `authprovider` import introduced in this PR relies on containerd 1.5.x
kube backend relies on helm -> deislabs/oras -> containerd 1.4.x

containerd updated `PushContent` signature to add a `platform` parameter. 
As Goland does not support polymorphism nor default parameter values (welcome into XXIth century) I can't find any possible workaround here 😱

File: go.mod [hunk 1]
```diff
@@ -37,40 +39,49 @@ require (
...
+	k8s.io/client-go v0.20.6
 	rsc.io/letsencrypt v0.0.3 // indirect
 	sigs.k8s.io/kustomize/kyaml v0.10.5
 )
 
 // (for buildx)
 replace github.com/jaguilar/vt100 => github.com/tonistiigi/vt100 v0.0.0-20190402012908-ad4c4a574305
+
+// see https://github.com/helm/helm/pull/9675
+replace helm.sh/helm/v3 => github.com/ndeloof/helm/v3 v3.1.0-rc.1.0.20210507084602-d0ceda767d95
```
Comment (gtardif) [inline go.mod:87]: A quick comment to explain this hack would be useful in the future...

**Review by gtardif — APPROVED** [review]
LGTM with nit (comment on go mod replace)

**Pull request merged**